src/resource/search/search.controller.ts

```diff
@@ -0,0 +1,36 @@
+import {
+    Controller,
+    Get,
+    Query,
+    UseGuards,
+} from '@nestjs/common';
+import { SearchService } from './search.service';
+import { JwtGuard } from '../../auth/guard';
+import { GetUser } from '../../decorators';
+import { UserEntity } from 'src/database/entities/user.entity';
+
+@UseGuards(JwtGuard)
+@Controller('search')
+export class SearchController {
+    constructor(private searchService: SearchService) {}
+
+    @Get('/all')
+    searchAll(@GetUser() user: UserEntity, @Query('query') query: string) {
+        return this.searchService.fetchAllBySearch(user, query);
+    }
+
+    @Get('/notes')
+    searchNotes(@GetUser() user: UserEntity, @Query('query') query: string) {
+        return this.searchService.fetchNotesBySearch(user, query);
+    }
+
+    @Get('/tasks')
+    searchTasks(@GetUser() user: UserEntity, @Query('query') query: string) {
+        return this.searchService.fetchTasksBySearch(user, query);
+    }
+
+    @Get('/events')
+    searchEvents(@GetUser() user: UserEntity, @Query('query') query: string) {
+        return this.searchService.fetchEventsBySearch(user, query);
+    }
```

Comment:
I think you should rather use @Param decorator to fetch not 'query' becase it is not saying anything. We just want to be clear so put GET parameter named for example 'search' and it will  be /all/:search. You can see this decorator in editNoteById in Note controller as an example.